tests/unit_tests/utils/test_general.py

```diff
@@ -206,3 +211,54 @@ def check_results(x_to_test, y_to_test, x_right, y_right, angle):
         gen.rotate(x_new_array.to(u.cm), y_new_array, angle_deg)
     with pytest.raises(u.core.UnitsError):
         gen.rotate(x_new_array, y_new_array, 30 * u.m)
+
+
+def test_convert_2D_to_radial_distr(caplog):
+
+    # Test normal functioning
+    max_dist = 100
+    num_bins = 100
+    step = max_dist / num_bins
+    xaxis = np.arange(-max_dist, max_dist, step)
+    yaxis = np.arange(-max_dist, max_dist, step)
+    x2d, y2d = np.meshgrid(xaxis, yaxis)
+    distance_to_center_2D = np.sqrt((x2d) ** 2 + (y2d) ** 2)
+
+    radial_edges, distance_to_center_1D = gen.convert_2D_to_radial_distr(
+        xaxis, yaxis, distance_to_center_2D, num_bins=num_bins, max_dist=max_dist
+    )
+    difference = radial_edges[:-1] - distance_to_center_1D
+
+    assert pytest.approx(difference[:-1], 1) == 0  # last value deviates
```

Comment:
Why do you have `, 1` in the `approx`? That essentially means this assertion would always pass cause you are setting `rel` option of `approx` to `1` (i.e., you allow a 100% error). It should be something like `pytest.approx(difference[:-1], 1e-6)`

Reply:
ok, I thought it is was similar to np.around but I am correcting this now then

Reply:
I found that for this specific line I only need to include the argument abs=1 and that is ok. I am fine with the accuracy of 1 meter. for the other pytest.approx ( in the other files, I will correct now)

Reply:
The step of the array is 1 meter, I would not expect any estimate to get better than that. if you check the value of the difference it is 0.2 ~ 0.5

Reply:
OK!